account_invoice_validate_send_email/models/account_invoice.py

```diff
@@ -67,10 +67,14 @@ def action_invoice_open(self):
         base_url = self.env["ir.config_parameter"].get_param("web.base.url")
         for invoice in self:
```

Comment:
```suggestion
        for invoice in self:
            if not invoice.type in ("out_invoice", "out_refund"):
                continue
```

Just in case. The current logic may already exclude the vendor bills, but we might as well make the intention clearer.

Reply:
Thank you, I add the code.
FYI, it seems that we have to write as follows according to the [pylint rules](https://www.flake8rules.com/rules/E713.html).

`
if invoice.type not in
`